mozci/push.py

```diff
@@ -459,11 +459,17 @@ def config_group_summaries(self):
             for group in task.groups:
                 config_groups[(task.configuration, group)].append(task)
 
-        return {
+        config_group_summaries = {
             config_group: GroupSummary(config_group[1], tasks)
             for config_group, tasks in config_groups.items()
         }
 
+        return {
+            config_group: summary
+            for config_group, summary in config_group_summaries.items()
+            if len(summary.tasks) > 0
+        }
```

Comment:
Rather than iterate over config_groups twice, I think it's more efficient and readable to use a for loop and build `config_group_summaries`.

Reply:
I usually prefer comprehensions, but I guess here the advantage is lost as you need two... I've applied the change.